packages/cli/src/upload/upload.js

```diff
@@ -542,7 +542,7 @@ async function runFilesystemTarget(options) {
     const fetchTimeDate = new Date(new Date(lhr.fetchTime).getTime() || Date.now());
     const context = {
       hostname: url.hostname,
-      pathname: url.pathname,
+      pathname: url.hash + url.pathname,
```

Comment:
we don't want to merge these two, we would want them to be separate
```suggestion
      pathname: url.pathname,
      hash: url.hash,
```

Reply:
@patrickhulce - Regarding the test case, we want to add the test case for both `collect` & `upload` or just the latter?

Reply:
Collect isn't being touched here, just upload

Reply:
@patrickhulce - Sounds good! I have made the suggested change and written the test case in a new file and it is working as expected. 

However, I have a question, NOT all the test cases are running on my machine. 
Refer following screenshot - 

The unit test case file, I have written is - `upload-url-hash.test.js`. The test for this has passed. 
<img width="611" alt="image" src="https://user-images.githubusercontent.com/20161217/197869325-0004b166-d71c-42b8-b3a8-6255f5a98661.png">

However, `NOT ALL` the test suites are passing. Refer following screenshots - 
<img width="930" alt="image" src="https://user-images.githubusercontent.com/20161217/197869503-147986bb-6d13-4ef3-9a60-76a680abba11.png">

Can I go ahead and commit the changes ? 
OR 
Do I need the entire test suite to pass?

Reply:
don't worry about the whole test suite for now, some are flaky and/or require other setup on your machine like mysql, if upload tests are passing you should be a-ok

Reply:
Perfect!

Reply:
@patrickhulce - Can you the checks again. The issue was related to syncing of the directories, I have added `rimraf` and the required code for the same. The test case is running successfully now, had 3 consecutive successful runs.

Reply:
@patrickhulce - Checked the raw logs for both `windows` and `macOS` checks that have failed, in that - the test case written for the PR has passed successfully.

The test case file name is - `upload-url-hash.test.js`.
Let me know, if there anything actionable at my end.